File: Runtime-位运算/Runtime-位运算/Student.m
//
//  Student.m
//  Runtime-位运算
//
//  Created by 赖永鹏 on 2019/3/25.
//  Copyright © 2019年 LYP. All rights reserved.
//

#import "Student.h"

@interface Student ()
{
//    位域
    struct{
        char tall:1;
        char high:1;
        char weight:1;
    }_tallHighWeight;
}

@end

@implementation Student

- (instancetype)init
{
    self = [super init];
    if (self) {

    }
    return self;
}
-(void)setTall:(BOOL)tall{
    
    _tallHighWeight.tall = tall;
}

-(void)setHigh:(BOOL)high{
    _tallHighWeight.high = high;
}

-(void)setWeight:(BOOL)weight{
    _tallHighWeight.weight = weight;
}

-(BOOL)isHigh{
    return !!_tallHighWeight.high;
}

-(BOOL)isWeight{
    return !!_tallHighWeight.weight;
}

-(BOOL)isTall{
    return !!_tallHighWeight.tall;
}
@end

File: Swift/基本语法/SwiftToOC/SwiftToOC/Student.swift
//
//  Student.swift
//  SwiftToOC
//
//  Created by 赖永鹏 on 2019/8/28.
//  Copyright © 2019年 LYP. All rights reserved.
//

import UIKit

@objcMembers class Student: Person {

    @objc dynamic var weight: Int = 0
    var observe:Observer = Observer()
    override init() {
        super.init()
        addObserver(observe, forKeyPath: "weight", options: [.new,.old], context: nil)
    }
    deinit {
        removeObserver(observe, forKeyPath: "weight")
    }
}
